### PR TITLE
feat: logs added for es8 index

### PR DESCRIPTION
**Pull request by aman-exp-infy**
Adding logs sequentially for all indexes in docarray according to [issue](https://github.com/docarray/docarray/issues/1462)

### CODE REVIEW

**Comment by aman-exp-infy** [comment]
`test_jac` failing https://github.com/docarray/docarray/actions/runs/5011253038/jobs/8981901725?pr=1551 @samsja @JohannesMessner

**Comment by agaraman0** [comment]
Hey @samsja I have tried to debug the failed test case but have not been able to figure it out as I have just added logs here.

**Comment by samsja** [comment]
these tests are failing because it needs to access a token in the docarray org but because you are opening a PR trough your fork you don't have access.

Basically don't worry we will merge even if it fails

**Comment by samsja** [comment]
I will review asap btw

File: docarray/index/backends/elastic.py [hunk 1]
```diff
@@ -139,8 +152,14 @@ def __init__(self, outer_instance, **kwargs):
                 'query': defaultdict(lambda: defaultdict(list))
             }
 
+            self._outer_instance._logger.debug('QueryBuilder is being initialized')
```
Comment (samsja) [inline docarray/index/backends/elastic.py:155]: in a first time lets not add logs to the query builder

Comment (agaraman0) [reply docarray/index/backends/elastic.py:155]: what do we mean by first time?

File: docarray/index/backends/elastic.py [hunk 2]
```diff
@@ -628,28 +684,42 @@ def _format_response(self, response: Any) -> Tuple[List[Dict], List[Any]]:
...
+        self._logger.debug(f'Refreshing the index {index_name}')
+
         self._client.indices.refresh(index=index_name)
 
     ###############################################
     # API Wrappers                                #
     ###############################################
 
     def _client_put_mapping(self, mappings: Dict[str, Any]):
+        self._logger.debug('Putting mapping to Elastic client')
```
Comment (samsja) [inline docarray/index/backends/elastic.py:696]: here you already logged when the function is called. Therefore it is redundant

**Review by samsja — CHANGES_REQUESTED** [review]
Lets not logger inside private function

File: docarray/index/backends/elastic.py [hunk 3]
```diff
@@ -104,29 +107,36 @@ def __init__(self, db_config=None, **kwargs):
...
         if default_index_name is None:
-            raise ValueError(
+            self._logger.error(
                 'A ElasticDocIndex must be typed with a Document type.'
                 'To do so, use the syntax: ElasticDocIndex[DocumentType]'
             )
-
-        return self._db_config.index_name or default_index_name
+            raise ValueError
+        index_name = self._db_config.index_name or default_index_name
```
Comment (samsja) [inline docarray/index/backends/elastic.py:137]: You can do smth like


error_msg = 'A ElasticDocIndex must be typed with a Document type.'
                'To do so, use the syntax: ElasticDocIndex[DocumentType]'

self._logger.error(error_msg)
raise ValueError(error_msg)

Comment (agaraman0) [reply docarray/index/backends/elastic.py:137]: oh okay.

**Review by samsja — APPROVED** [review]

**Pull request merged**